opal/rocm/common_rocm.c

```diff
@@ -0,0 +1,319 @@
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipMemcpy) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipMemcpyAsync", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipMemcpyAsync\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipMemcpyAsync) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipStreamCreate", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipStreamCreate\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipStreamCreate) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipStreamDestroy", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipStreamDestroy\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipStreamDestroy) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipStreamSynchronize", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipStreamSynchronize\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipStreamSynchronize) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipGetErrorString", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipGetErrorString\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipGetErrorString) = ptr;
+
+    ret = opal_dl_lookup (hip_handle, "hipPointerGetAttributes", &ptr, &str);
+    if (OPAL_SUCCESS != ret) {
+        opal_output(0, "Failed to find hipPointerGetAttributes\n");
+        dlclose(hip_handle);
+        return OPAL_ERROR;
+    }
+    *(void **) (&hip_funcs.hipPointerGetAttributes) = ptr;
+
+    return OPAL_SUCCESS;
+}
+
+static void hip_dl_finalize(void)
+{
+    hipError_t err = hip_funcs.hipStreamDestroy(&common_rocm_MemcpyStream);
+    if (hipSuccess != err) {
+        opal_output(0, "hip_dl_finalize: error while destroying the hipStream\n");
+    }
+    dlclose(hip_handle);
+}
+
+
+static void mca_common_rocm_support_init(void)
+{
+    if (rocm_initialized) {
+        return;
+    }
+
+    OBJ_CONSTRUCT(&common_rocm_init_lock, opal_mutex_t);
+    OPAL_THREAD_LOCK(&common_rocm_init_lock);
+
+    if (rocm_initialized) {
+        return;
+    }
+
+    if (!OPAL_HAVE_DL_SUPPORT) {
+        opal_output(0, "Can not open libamdhip64.so without dlopen support\n");
+        OPAL_THREAD_UNLOCK(&common_rocm_init_lock);
+        return;
+    }
+    if (OPAL_SUCCESS != hip_dl_init()) {
+        opal_output(0, "Could not open libamdhip64.so. Please check your LD_LIBRARY_PATH\n");
+        OPAL_THREAD_UNLOCK(&common_rocm_init_lock);
+        return;
+    }
+    rocm_initialized = true;
+    opal_rocm_runtime_initialized = true;
+
+    mca_common_rocm_register_mca_variables();
```

Comment:
Yes. this is understood and even mentioned in the pr description above :-) First, I am open to register the variables somewhere else, I just need  a pointer where to do that, since there is no rocm component at the moment. The discussion might however be moot if we move to use the accelerator framework.

Reply:
Moved the invokation of the mca_common_rocm_register_mca_variables function for the time being to opal_init_utils, this way it shows up in ompi_info. Will remove it from here again once we move the code to the accelerator framework.